packages/service-library/src/servicelib/aiohttp/requests_validation.py

```diff
@@ -154,11 +154,13 @@ def parse_request_query_parameters_as(
         use_error_v1=use_enveloped_error_v1,
     ):
         data = dict(request.query)
-        return parameters_schema.parse_obj(data)
+        if hasattr(parameters_schema_cls, "parse_obj"):
+            return parameters_schema_cls.parse_obj(data)
+        return parse_obj_as(parameters_schema_cls, data)
```

Comment:
Question: can't you just use the `parse_obj_as` function call in both cases?

Reply:
there is a difference. In ``parse_obj_as`` there is an overhead that checks types and creates a new model class dynamically. I want to avoid that when a class is a BaseModel